src/App.cpp

```diff
@@ -286,23 +331,137 @@ void App::_drawControls()
+                    _headphones->changingAsmLevel.desired = ImGui::IsItemActive()
+                        ? THMSGV2T1::ValueChangeStatus::UNDER_CHANGING : THMSGV2T1::ValueChangeStatus::CHANGED;
+                    if (supportsAutoAsm) {
+                        ImGui::Checkbox("Auto Ambient Sound", &_headphones->autoAsmEnabled.desired);
+
+                        ImGui::BeginDisabled(!_headphones->autoAsmEnabled.current);
+                        static const std::map<THMSGV2T1::NoiseAdaptiveSensitivity, const char*> NoiseAdaptiveSensitivity_STR = {
+                            {THMSGV2T1::NoiseAdaptiveSensitivity::STANDARD, "Standard"},
+                            {THMSGV2T1::NoiseAdaptiveSensitivity::HIGH, "High"},
+                            {THMSGV2T1::NoiseAdaptiveSensitivity::LOW, "Low"},
+                        };
+                        auto it = NoiseAdaptiveSensitivity_STR.find(_headphones->autoAsmSensitivity.current);
+                        const char* currentStr = it != NoiseAdaptiveSensitivity_STR.end() ? it->second : "Unknown";
+                        if (ImGui::BeginCombo("Sensitivity", currentStr)) {
+                            for (auto const& [k, v] : NoiseAdaptiveSensitivity_STR) {
+                                bool is_selected = k == _headphones->autoAsmSensitivity.current;
+                                if (ImGui::Selectable(v, is_selected))
+                                    _headphones->autoAsmSensitivity.desired = k;
+                                if (is_selected)
+                                    ImGui::SetItemDefaultFocus();
+                            }
+
+                            ImGui::EndCombo();
+                        }
+
+                        ImGui::EndDisabled();
+                    }
+
+                    ImGui::Checkbox("Voice Passthrough", &_headphones->asmFoucsOnVoice.desired);
+                }
+
                 ImGui::EndTabItem();
             }
 
-            if (ImGui::BeginTabItem("Speak to Chat")) {
+            if (_headphones->supports(MessageMdrV2FunctionType_Table1::SMART_TALKING_MODE_TYPE2)
+                && ImGui::BeginTabItem("Speak to Chat")) {
                 ImGui::Checkbox("Enabled", &_headphones->stcEnabled.desired);
                 ImGui::SliderInt("STC Level", &_headphones->stcLevel.desired, 0, 2);
                 ImGui::SliderInt("STC Time", &_headphones->stcTime.desired, 0, 3);
                 ImGui::EndTabItem();
             }
 
+            if (_headphones->supports(MessageMdrV2FunctionType_Table1::LISTENING_OPTION) &&
+                ImGui::BeginTabItem("Listening Mode")) {
+                ListeningMode effectiveMode = _headphones->listeningModeConfig.current.getEffectiveMode();
+
+                // Standard
+                bool radioChanged = false;
+                if (radioChanged |= ImGui::RadioButton("Standard", effectiveMode == ListeningMode::Standard))
```

Comment:
Just realized that `if (radioChanged |= ImGui::RadioButton(...)) ...` is wrong. Should've been `if (ImGui::RadioButton(...)) { radioChanged |= true; ... }`

I'll address this in an upcoming PR.